src/main/java/net/minecrell/terminalconsole/TerminalConsoleAppender.java

```diff
@@ -205,44 +206,48 @@ protected TerminalConsoleAppender(String name, Filter filter, Layout<? extends S
         initializeTerminal();
     }
 
-    private synchronized static void initializeTerminal() {
```

Comment:
I think I would prefer if you keep this method as-is and rather change the calls of `initializeTerminal();` to

```java
if (!initialized)
    initializeTerminal();
```

AFAICT there are just two calls of `initializeTerminal();` so the code overhead is negligible. I personally don't like the explicit `synchronized (TerminalConsoleAppender.class) {` very much.

Reply:
It is used more in HighlightErrorConverter, but sure, the overhead should still be very small

Reply:
`HighlightErrorConverter` doesn't call `initializeTerminal()` directly though. What I mean is that you change `isAnsiSupported()` to

```java
public static boolean isAnsiSupported() {
        if (!initialized)
             initializeTerminal();
        return ANSI_OVERRIDE != null ? ANSI_OVERRIDE : terminal != null;
    }
```

and same for the constructor